.github/workflows/test.yml

```diff
@@ -6,42 +6,99 @@ on:
       - "master"
   pull_request:
     branches:
-      - "*"
+      - "**"
   workflow_dispatch:
   release:
     types: [published, edited]
 
 jobs:
   tests:
-
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
         python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
 
     steps:
-    - uses: actions/checkout@v4
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v5
-      with:
-        python-version: ${{ matrix.python-version }}
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        python -m pip install tox tox-gh-actions
-    - name: Cache tox environments
-      id: cache-tox
-      uses: actions/cache@v4
-      with:
-        path: .tox
-        # setup.py and setup.cfg have versioning info that would impact the
-        # tox environment. hashFiles only takes a single file path or pattern
-        # at the moment.
-        key: ${{ runner.os }}-${{ matrix.python-version }}-tox-${{ hashFiles('setup.py') }}-${{ hashFiles('setup.cfg') }} }}
-    - name: Test with tox
-      run: tox
-    - uses: codecov/codecov-action@v4
-      with:
-        token: ${{ secrets.CODECOV_TOKEN }}
-        verbose: true
+      - uses: actions/checkout@v4
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install tox tox-gh-actions
+      - name: Cache tox environments
+        id: cache-tox
+        uses: actions/cache@v4
+        with:
+          path: .tox
+          # setup.py and setup.cfg have versioning info that would impact the
+          # tox environment. hashFiles only takes a single file path or pattern
+          # at the moment.
+          key: ${{ runner.os }}-${{ matrix.python-version }}-tox-${{ hashFiles('setup.py') }}-${{ hashFiles('setup.cfg') }} }}
+      - name: Test with tox
+        run: tox
+      - uses: codecov/codecov-action@v4
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          verbose: true
+
+  # Always run this last as it can push new changes and actions will not rerun.
+  # not working on pull requests from forks
```

Comment:
Any suggestion on how to circumvent this? I mean it is pretty neat 👌  and could be helpful for external/new contributors, but if it doesn't work for them anyway I'm not sure of the utility (I will personally "always" use pre-commit locally :))

Reply:
i know i don't always remember to run pre-commit, but I use other ways to enforce it which are not practical here. 
I've decided to do it differently (run inside tox for one env only and have a step to commit changes if they exist)

Reply:
Oh, ok, I would always run it as an installed pre-commit hook, so not running it would be an explicit opt-out every time (`git commit --no-verify`)